src/incubator/Slider/index.tsx

```diff
@@ -235,37 +237,45 @@ const Slider = React.memo((props: Props) => {
   const _thumbStyle = useSharedValue(StyleUtils.unpackStyle(customThumbStyle || defaultThumbStyle, {flatten: true}));
   const _activeThumbStyle = useSharedValue(StyleUtils.unpackStyle(activeThumbStyle, {flatten: true}));
 
-  useEffect(() => {
-    if (!thumbStyle) {
-      _thumbStyle.value = StyleUtils.unpackStyle(defaultThumbStyle, {flatten: true});
-    }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [defaultThumbStyle, thumbStyle]);
-
-  useImperativeHandle(forwardedRef, () => ({
-    reset: () => reset()
-  }));
-
-  const reset = () => {
-    setInitialPositions(trackSize.value.width);
-    onReset?.();
-  };
-
-  const setInitialPositions = (trackWidth: number) => {
+  const setInitialPositions = useCallback((trackWidth: number) => {
     validateValues(props);
 
-    const defaultThumbPosition = getOffsetForValue(
-      useRange ? initialMinimumValue : value,
+    const defaultThumbPosition = getOffsetForValue(useRange ? initialMinimumValue : value,
       trackWidth,
       minimumValue,
       maximumValue);
     const rangeThumbPosition = getOffsetForValue(initialMaximumValue, trackWidth, minimumValue, maximumValue);
     defaultThumbOffset.value = defaultThumbPosition;
     rangeThumbOffset.value = useRange ? rangeThumbPosition : trackWidth;
+  }, [value]);
+
+  const reset = () => {
+    setInitialPositions(trackSize.value.width);
+    onReset?.();
   };
 
+  useImperativeHandle(forwardedRef, () => ({
+    reset: () => reset()
+  }));
+  
+  useEffect(() => {
+    didValueUpdate.current = true;
+    setInitialPositions(trackSize.value.width);
+  }, [value, setInitialPositions]);
+
+  useEffect(() => {
+    if (!thumbStyle) {
+      _thumbStyle.value = StyleUtils.unpackStyle(defaultThumbStyle, {flatten: true});
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [defaultThumbStyle, thumbStyle]);
+
   const onValueChangeThrottled = useCallback(_.throttle(value => {
-    onValueChange?.(value);
+    if (!didValueUpdate.current) { // don't invoke onChange when slider's value changes to prevent updates loop
```

Comment:
```suggestion
    if (!didValueUpdate.current) { // NOTE: fix for GradientSlider (should be removed after fix in the GradientSlider component): don't invoke onChange when slider's value changes to prevent updates loop
```